docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,197 @@
+2. Referral Manager receives all userIDs from satellites and then generates x amount of invitation tokens per user based on the input from CLI.
+3. Referral Manager adds the newly generated invitation tokens and the users each token associated with into Referral Manager database.
+4. After storing tokens into the database, Referral Manager sends invitation tokens along with the owner IDs back to corresponding host satellites.
+5. Host satellite receives the data and then stores the invitation tokens into `registration_token` table so they can be displayed on the satellite GUI
+
+_Referral Link Redemption_
+
+1. User Alice tries to register a new account through a referral link, which triggers redeemed satellite to verify invitation token using Referral manager.
+2. Referral Manager checks the status of the token:
+    - if it is not redeemed, Referral Manager sends back a success response to the redeemed satellite and mark the token as redeemed in the Referral Manager's database
+    - if it token is already redeemed, Referral Manager sends back a `invalid token` response to the redeemed satellite.
+3. Redeemed satellite receives the response, which:
+    - if it is a success, the satellite will proceed with the account creation and
+    then send a request to the Referral Manager to save the newly created user ID along with the used token.
+    - if it is an invalid token, the satellite will display a proper message in the UI.
+4. Referral Manager reaches out to the host satellite holding the redeemed token, so that it can be removed from the UI.
+    
+_User Interface For Displaying Referral Link_
+
+1. Users will have a referral tab on the UI. On click, it will try to retrieve referral tokens from the backend:
+    - if the front-end receives tokens, it should display them.
+    - if the front-end receives empty response payload, it should display a message `No available referral link. Try again later.`
+
+## Rationale
+
+We could add a new column `referral_tokens` in `users` table to store user's unredeemed tokens, since the `registration_token` is a temporary table.
+
+## Implementation
+
+- Create a private repository for Referral Manager.
+- Create `tokens` and `satellites` table in Referral Manager database.
+- Create a `Delete` method for satellite `registration_tokendb`.
+- Implementing an endpoint on satellite for gathering users whose current count of remaining invitation token is 0.
+- Implementing generating invitation tokens from Referral Manager CLI.
+- Implementing an endpoint on satellite for saving new referral links into users table.
+- Implementing an endpoint on Referral Manager for verifying invitation tokens.
+- Replace existing registration token logic.
+
+Pseudocode
+
+The Token struct represents both a Go struct (on satellite and Referral Manager) as well as the schema for the `tokens` table on the Referral Manager
+
+```
+type Token struct {
+	Secret [32]byte
+	OwnerID uuid.UUID
+	RedeemedBy uuid.UUID
```

Comment:
Thoughts on changing `RedeemedBy` to `RedeemedID` to make it more consistent with `OwnerID`?